compiler/plugins/input/StableHLO/stablehlo-iree/Conversion/StableHLOCustomCalls.cpp

```diff
@@ -0,0 +1,244 @@
+
+  auto outMap = b.getMultiDimIdentityMap(hShape.size());
+
+  llvm::SmallVector<AffineExpr> exprs;
+  for (int i = 0; i < vTy.getRank(); ++i) {
+    exprs.push_back(b.getAffineDimExpr(i));
+  }
+
+  AffineMap vMap = AffineMap::get(hTy.getRank(), 0, exprs, b.getContext());
+
+  exprs.back() = b.getAffineDimExpr(vTy.getRank());
+  AffineMap vTMap = AffineMap::get(hTy.getRank(), 0, exprs, b.getContext());
+
+  SmallVector<AffineMap> affineMaps = {vMap, vTMap, outMap};
+
+  SmallVector<utils::IteratorType> iterTypes(hShape.size(),
+                                             utils::IteratorType::parallel);
+
+  Value zero = b.create<arith::ConstantOp>(b.getZeroAttr(vTy.getElementType()));
+  Value one =
+      b.create<arith::ConstantOp>(b.getFloatAttr(vTy.getElementType(), 1.0));
+
+  return b
+      .create<linalg::GenericOp>(
+          hTy, ValueRange{v, v}, empty, affineMaps, iterTypes,
+          [&](OpBuilder &bb, Location loc, ValueRange args) {
+            ImplicitLocOpBuilder b(loc, bb);
+            llvm::SmallVector<Value> indices;
+            for (int i = 0; i < hTy.getRank(); ++i) {
+              indices.push_back(b.create<linalg::IndexOp>(loc, i));
+            }
+
+            llvm::SmallVector<Value> tauIndices(indices.begin(),
+                                                indices.end() - 2);
+            tauIndices.push_back(k);
+            Value t = b.create<tensor::ExtractOp>(tau, tauIndices);
+
+            auto tri = [&](Value v, Value i) {
+              Value eq =
+                  b.create<arith::CmpIOp>(arith::CmpIPredicate::eq, i, k);
+              Value lt =
+                  b.create<arith::CmpIOp>(arith::CmpIPredicate::ult, i, k);
+              Value sel = b.create<arith::SelectOp>(eq, one, v);
+              return b.create<arith::SelectOp>(lt, zero, sel);
+            };
+
+            Value v = tri(args[0], indices[indices.size() - 2]);
+            Value vT = tri(args[1], indices[indices.size() - 1]);
+
+            Value h = b.create<arith::MulFOp>(v, vT);
+            h = b.create<arith::MulFOp>(h, t);
+
+            Value isDiag = b.create<arith::CmpIOp>(arith::CmpIPredicate::eq,
+                                                   indices[indices.size() - 2],
+                                                   indices[indices.size() - 1]);
+            Value diag = b.create<arith::SelectOp>(isDiag, one, zero);
+            Value sub = b.create<arith::SubFOp>(diag, h);
+
+            b.create<linalg::YieldOp>(sub);
+          })
+      .getResult(0);
+}
+
+// Compute the householder matrice with the k-th column of matrix and k-th tau
+// value.
+static Value computeHouseholderSlice(Value matrix, Value tau, Value k,
+                                     ImplicitLocOpBuilder b) {
+  auto matrixTy = cast<ShapedType>(matrix.getType());
+  int rank = matrixTy.getRank();
+
+  llvm::SmallVector<OpFoldResult> vStrides(rank, b.getIndexAttr(1));
```

Comment:
Is llvm:: required here? You are in mlir namespace and some of these are forwarded via support to that namespace.

Reply:
Nope, not required. Removed.